adobe/spectrum-css
# refactor(datepicker)!: diy token migration

## Description

Migrates DatePicker to new tokens. 
DatePicker does not have a figma or XD files, however it is similar to Combo box which does. 

Additionally: 
- addresses an issue with the invalid icons being hidden behind the picker button (see screenshots) 
- ensures WHCM 
- add quiet variant to storybook
- as per design feedback, border of the picker button has been updated to match the border of text-field 


[Jira Ticket 579](https://jira.corp.adobe.com/browse/CSS-579)



## How and where has this been tested?

Please tag yourself on the tests you've marked complete to confirm the tests have been run by someone other than the author.



### Testing Outline 
@jawinn
1. open the [docs site](https://pr-2185--spectrum-css.netlify.app/datepicker) for the DatePicker component: 
- [x] with the exception of the invalid icon positioning, styling should match production 
- [x] Invalid icon should be positioned to the left of the picker button and match styling of invalid icon on ComboBox 

2. open the [storybook](https://pr-2185--spectrum-css.netlify.app/preview/?path=/docs/components-date-picker--docs) for the DatePicker component
- [x] default and quiet variants are displayed 
- [x] controls work as expected

### Regression testing

Validate:

1. A legacy documentation page (such as [accordion](https://pr-###--spectrum-css.netlify.app/accordion.html)), including:


4. A migrated documentation page (such as [action group](https://pr-###--spectrum-css.netlify.app/actiongroup.html)), including:


## Screenshots

### Before
![Screenshot 2023-09-28 at 1 19 54 PM](https://github.com/adobe/spectrum-css/assets/63808889/c677047e-a3c8-4e2f-927e-379248e83906)
![Screenshot 2023-09-28 at 1 20 02 PM](https://github.com/adobe/spectrum-css/assets/63808889/4ceea6c1-0031-4efb-b990-02a83f08d60d)


### After
![Screenshot 2023-10-07 at 8 39 55 AM](https://github.com/adobe/spectrum-css/assets/63808889/e35014b6-dd45-46dd-bb87-379a5ab35b18)
![Screenshot 2023-10-07 at 8 40 04 AM](https://github.com/adobe/spectrum-css/assets/63808889/e9e00dd7-f61e-45d7-81ae-0ab5891364be)




## To-do list
